### PR TITLE
Fix copy code snippet to not remove spread operators

### LINKED ISSUE #105: Fix copy code JS, to only remove ellipsis at the start of the line
Otherwise, it will remove the spread operator from within code too. I.e. `...Permissions.default(),` becomes `Permissions.default(),`.

### PR DESCRIPTION
**Pull request by ymekuria**
Fixes #105

### CODE REVIEW

File: src/theme/CodeBlock/CopyButton/index.js [hunk 1]
```diff
@@ -9,16 +9,21 @@ export default function CopyButton({ code, className }) {
   const copyTimeout = useRef(undefined);
   const handleCopyCode = useCallback(() => {
     const removeLineNumbers = /^\s*\d+\s/gm;
-    const removeEllipses = /\.{3}/gm;
+    const removeEllipses = /(^\s*\.{3})(?![^a-z])|(^\.{3})/gm;
```
Comment (jasongitmail) [inline src/theme/CodeBlock/CopyButton/index.js:12]: @ymekuria What's the first part for? I'd have thought your second part would solve it `(^\.{3})`

**Review by jasongitmail — APPROVED** [review]
Tested and works well. Thanks!

File: src/theme/CodeBlock/CopyButton/index.js (hunk 1 above)
Comment (ymekuria) [reply src/theme/CodeBlock/CopyButton/index.js:12]: In testing I found that sometimes the elipses have preceding spaces. The first part ensures that spread operators are not removed when matching elipses that have preceding spaces.

Comment (jasongitmail) [reply src/theme/CodeBlock/CopyButton/index.js:12]: cool. thanks for explaining

**Pull request merged**